nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/flow-designer/state/flow/flow.effects.ts

```diff
@@ -348,14 +352,78 @@ export class FlowEffects {
                         this.flowService.goToRemoteProcessGroup(request);
                     } else {
                         this.store.dispatch(
-                            showOkDialog({ title: 'Remote Process Group', message: 'No target URI defined.' })
+                            FlowActions.showOkDialog({
+                                title: 'Remote Process Group',
+                                message: 'No target URI defined.'
+                            })
                         );
                     }
                 })
             ),
         { dispatch: false }
     );
 
+    startRemoteProcessGroupPolling$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(FlowActions.startRemoteProcessGroupPolling),
+            switchMap((request) => {
+                return interval(3000, asyncScheduler).pipe(
+                    takeUntil(this.actions$.pipe(ofType(FlowActions.stopRemoteProcessGroupPolling)))
+                );
+            }),
+            switchMap(() => {
+                return of(FlowActions.refreshRemoteProcessGroup());
+            })
+        )
+    );
+
+    requestRemoteProcessGroupPolling$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(FlowActions.requestRemoteProcessGroupPolling),
+            switchMap(() => {
+                return of(FlowActions.startRemoteProcessGroupPolling());
+            })
+        )
+    );
+
+    refreshRemoteProcessGroup$ = createEffect(() =>
+        this.actions$.pipe(
+            ofType(FlowActions.refreshRemoteProcessGroup),
+            concatLatestFrom(() => this.store.select(selectRpgToPoll)),
+            map(([, rpgToPoll]) => {
+                return rpgToPoll;
+            }),
+            isDefinedAndNotNull(),
```

Comment:
I think this can be replaced with

`concatLatestFrom(() => this.store.select(selectRpgToPoll).pipe(isDefinedAndNotNull()),`